### ISSUE TITLE
Improve .npmignore

### ISSUE THREAD

**Issue by SimonSchick** [Esri/terraformer#308]
You are shipping a HUGE amount of redundant files to npm:

```
node_modules/terraformer/.travis.yml
node_modules/terraformer/CHANGELOG.md
node_modules/terraformer/Gemfile
node_modules/terraformer/Gemfile.lock
node_modules/terraformer/bower.json
node_modules/terraformer/gruntfile.js
node_modules/terraformer/package.json
node_modules/terraformer/release.sh

node_modules/terraformer/test.js
node_modules/terraformer/test.ts
node_modules/terraformer/tsconfig.json
node_modules/terraformer/tslint.json
node_modules/terraformer/.coverage/
node_modules/terraformer/.coverage/base.css
node_modules/terraformer/.coverage/coverage.json
node_modules/terraformer/.coverage/index.html
node_modules/terraformer/.coverage/prettify.css
node_modules/terraformer/.coverage/prettify.js
node_modules/terraformer/.coverage/sort-arrow-sprite.png
node_modules/terraformer/.coverage/sorter.js
node_modules/terraformer/.coverage/__root__/
node_modules/terraformer/.coverage/__root__/index.html
node_modules/terraformer/.coverage/__root__/terraformer.js.html
node_modules/terraformer/docs-build/
node_modules/terraformer/docs-build/CNAME
node_modules/terraformer/docs-build/favicon.ico
node_modules/terraformer/docs-build/index.html
node_modules/terraformer/docs-build/arcgis-parser/
node_modules/terraformer/docs-build/arcgis-parser/index.html
node_modules/terraformer/docs-build/assets/
node_modules/terraformer/docs-build/assets/css/
node_modules/terraformer/docs-build/assets/css/terraformer-02847305.css
node_modules/terraformer/docs-build/assets/fonts/
node_modules/terraformer/docs-build/assets/fonts/esri-logo-3a94862e.svg
node_modules/terraformer/docs-build/assets/fonts/esri-logo-71163913.ttf
node_modules/terraformer/docs-build/assets/fonts/esri-logo-80bed756.woff
node_modules/terraformer/docs-build/assets/fonts/esri-logo-fc93d7bf.eot
node_modules/terraformer/docs-build/assets/fonts/esri-logo.dev-aed9f839.svg
node_modules/terraformer/docs-build/assets/images/
node_modules/terraformer/docs-build/assets/images/terraformer-arcparser-a0228324.png
node_modules/terraformer/docs-build/assets/images/terraformer-core-93b83c36.png
node_modules/terraformer/docs-build/assets/images/terraformer-geostore-9e09ce2c.png
node_modules/terraformer/docs-build/assets/images/terraformer-parser-a57c9c2d.png
node_modules/terraformer/docs-build/assets/images/terraformer-wktparser-b18bc9d9.png
node_modules/terraformer/docs-build/assets/javascripts/
node_modules/terraformer/docs-build/assets/javascripts/all-da39a3ee.js
node_modules/terraformer/docs-build/assets/javascripts/classie-b6db1f70.js
node_modules/terraformer/docs-build/assets/javascripts/drawer-3a1490eb.js
node_modules/terraformer/docs-build/assets/javascripts/modernizr.custom-cadc78a2.js
node_modules/terraformer/docs-build/core/
node_modules/terraformer/docs-build/core/index.html
node_modules/terraformer/docs-build/geostore/
node_modules/terraformer/docs-build/geostore/index.html
node_modules/terraformer/docs-build/geostore/alternate-indexes/
node_modules/terraformer/docs-build/geostore/alternate-indexes/index.html
node_modules/terraformer/docs-build/geostore/core-concepts/
node_modules/terraformer/docs-build/geostore/core-concepts/index.html
node_modules/terraformer/docs-build/geostore/data-stores/
node_modules/terraformer/docs-build/geostore/data-stores/index.html
node_modules/terraformer/docs-build/geostore/spatial-indexes/
node_modules/terraformer/docs-build/geostore/spatial-indexes/index.html
node_modules/terraformer/docs-build/getting-started/
node_modules/terraformer/docs-build/getting-started/index.html
node_modules/terraformer/docs-build/glossary/
node_modules/terraformer/docs-build/glossary/index.html
node_modules/terraformer/docs-build/install/
node_modules/terraformer/docs-build/install/index.html
node_modules/terraformer/docs-build/partials/
node_modules/terraformer/docs-build/partials/cover/
node_modules/terraformer/docs-build/partials/cover/index.html
node_modules/terraformer/docs-build/partials/doctoc/
node_modules/terraformer/docs-build/partials/doctoc/index.html
node_modules/terraformer/docs-build/partials/footer/
node_modules/terraformer/docs-build/partials/footer/index.html
node_modules/terraformer/docs-build/partials/index_partials/
node_modules/terraformer/docs-build/partials/index_partials/arcgis_parser/
node_modules/terraformer/docs-build/partials/index_partials/arcgis_parser/index.html
node_modules/terraformer/docs-build/partials/index_partials/geostore/
node_modules/terraformer/docs-build/partials/index_partials/geostore/index.html
node_modules/terraformer/docs-build/partials/index_partials/terraformer_core/
node_modules/terraformer/docs-build/partials/index_partials/terraformer_core/index.html
node_modules/terraformer/docs-build/partials/index_partials/wkt_parser/
node_modules/terraformer/docs-build/partials/index_partials/wkt_parser/index.html
node_modules/terraformer/docs-build/partials/nav/
node_modules/terraformer/docs-build/partials/nav/index.html
node_modules/terraformer/docs-build/partials/subnav/
node_modules/terraformer/docs-build/partials/subnav/index.html
node_modules/terraformer/docs-build/wkt-parser/
node_modules/terraformer/docs-build/wkt-parser/index.html
node_modules/terraformer/spec/
node_modules/terraformer/spec/geojsonHelpers.js
node_modules/terraformer/spec/spatialReferenceSpec.js
node_modules/terraformer/spec/terraformerSpec.js
```

Please add those to prevent package size bloat, this should be a patch release.

**Comment by jgravois** [comment]
thanks for the report. my preference would be to remove the `.npmignore` and use [`pkg.files`](https://docs.npmjs.com/files/package.json#files) instead.

if no one else wants to help, i'll find time to take care of this soon.

**Comment by SimonSchick** [comment]
Both solutions are fine, I tend to prefer using an inverted glob for npm ingore files such as:

```
/*
!/dist
```

essential files are automatically whitelisted by npm!

**Comment by jgravois** [comment]
released as: [`v1.0.9`](https://unpkg.com/terraformer@1.0.9/) :tada:

Name | Type | Size | Last Modified
-- | -- | -- | --
CHANGELOG.md | text/markdown | 2.65 kB | 1985-10-26T08:15:00.000Z
LICENSE | text/plain | 1.06 kB | 1985-10-26T08:15:00.000Z
README.md | text/markdown | 3.94 kB | 1985-10-26T08:15:00.000Z
package.json | application/json | 1.3 kB | 1985-10-26T08:15:00.000Z
terraformer.d.ts | text/plain | 14.3 kB | 1985-10-26T08:15:00.000Z
terraformer.js | application/javascript | 38.44 kB | 1985-10-26T08:15:00.000Z
